erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -737,7 +737,7 @@ void WebRtcConnection::syncWrite(std::shared_ptr<DataPacket> packet) {
 }
 
 void WebRtcConnection::setTransport(std::shared_ptr<Transport> transport) {  // Only for Testing purposes
-  video_transport_ = transport;
+  video_transport_ = std::move(transport);
```

Comment:
In general I'm a bit reluctant to use std::move() in places that aren't hot paths because it's too easy to introduce bugs if we try to use the pointer later in the code, am I right?

Reply:
Coping the shared pointer basically causes reference counter increment, function move instead doesn't affect reference counter at all because it simply transfers its ownership to another object.
Please consider the following code:

```
#include <iostream>
#include <memory>

class A
{
 public:
  std::string some_usefull_string;
};

void do_stuff_with_copied_shared_ptr(std::shared_ptr<A> ptr)
{
  std::cout << "before coping use_count is " << ptr.use_count() << std::endl;
  std::shared_ptr<A> local_ptr = ptr;
  std::cout << "after coping use_count is " << ptr.use_count() << std::endl;
}

void do_stuff_with_moved_shared_ptr(std::shared_ptr<A> ptr)
{
  std::cout << "before moving use_count is " << ptr.use_count() << std::endl;
  std::shared_ptr<A> local_ptr = std::move(ptr);
  std::cout << "after moving ptr use_count is " << ptr.use_count() << std::endl;
  if(!ptr)
    std::cout << "because ptr is not valid any more" << std::endl;
  if(local_ptr)
    std::cout << "but local_ptr is valid" << std::endl;
  std::cout << "and local_ptr use_count is " << local_ptr.use_count() << std::endl;
}

int main()
{
  std::shared_ptr<A> ptr = std::make_shared<A>();
  std::cout << "after construction use_count is " << ptr.use_count() << std::endl;
  do_stuff_with_copied_shared_ptr(ptr);
  do_stuff_with_moved_shared_ptr(ptr);
  std::cout << "before leaving use_count is " << ptr.use_count() << std::endl;
  return 0;
}

```
Obviously std::move() invalidates the object passed (ptr in this case), so you should be aware of this, but you can use "local_ptr" which is always valid. And this is the best practice in small functions like setters because the reference counter is atomic and changing it requires more time then a for standard variable. It's just a little performance improvement, so fill you free to use it in the way you like. :)
For more info please take a look at [this topic](https://stackoverflow.com/questions/41871115/why-would-i-stdmove-an-stdshared-ptr
) on stackoverflow.

Reply:
Yes, I know how it works, my point is whether the possibility to introduce bugs if we use an invalidated pointer when updating this code by accident worths the code optimization. We always have to consider that specially in OS projects we can receive PRs updating this code and it could be difficult to detect this kind of issues during a review. But I also see that setters are a reasonable place where we can use std::move.

Reply:
I perfectly agree. So if it's OK for you, in my PRs I'll move(shared_ptr) only in one line setters.